src/app/src/main/kotlin/org/imperial/mrc/hint/security/HintPac4jConfigService.kt

```diff
@@ -0,0 +1,40 @@
+package org.imperial.mrc.hint.security
+
+import org.imperial.mrc.hint.db.UserRepository
+import org.imperial.mrc.hint.security.oauth2.clients.HintClientsContext
+import org.imperial.mrc.hint.security.oauth2.clients.OAuth2Client
+import org.pac4j.core.client.Clients
+import org.pac4j.core.client.IndirectClient
+import org.pac4j.core.config.Config
+import org.pac4j.core.http.callback.PathParameterCallbackUrlResolver
+import org.pac4j.http.client.indirect.FormClient
+import org.pac4j.jee.context.session.JEESessionStore
+import org.pac4j.sql.profile.service.DbProfileService
+
+class HintPac4jConfigService(
```

Comment:
This class just reproduces the logic that was already in SecurityConfig.getPac4jConfig? Is this just an opportunistic refactor then, rather than necessary for the SSO changes?

Reply:
Opportunistic refactor, should have commented on it.